lib/ruby_indexer/lib/ruby_indexer/configuration.rb

```diff
@@ -6,15 +6,33 @@ class Configuration
     extend T::Sig
 
     sig { returns(T::Array[String]) }
-    attr_accessor :files_to_index
+    attr_accessor :excluded_gems
+
+    sig { returns(T::Array[String]) }
+    attr_accessor :excluded_patterns
 
     sig { void }
     def initialize
+      @excluded_gems = T.let([], T::Array[String])
+      @excluded_patterns = T.let(["*_test.rb"], T::Array[String])
```

Comment:
This seems too arbitrary. What about files named `something_spec.rb`?

Reply:
It is already. You can just do
```ruby
# .indexrc

RubyIndexer.configure do |config|
  config.excluded_patterns << "**/spec/**/*_spec.rb"
end
```

Given that we default to Minitest, I think it's reasonable to provide that as the default. People can configure the paths in any way their prefer.

Reply:
Not necessarily all projects need to have one. But if people want to tune the index for better accuracy and performance, they can. For most projects, just indexing the defaults will probably be fast enough and not require a lot of configs. So if the file doesn't exist, we just use the default configurations.

For super large projects, it will probably be worth excluding gems that don't make sense to be indexed. Or if the project has a different test setup, it can exclude them with a different pattern.

I like the default values approach a little better. You get a base configuration and can just tune what you need.

Reply:
If I have a [fixture class](https://github.com/rails/rails/blob/main/activerecord/test/activejob/job_runtime_test.rb#L7) in a test file, could I still use go-to-definition to it? It looks like I wouldn't be able to do that even if I'm in the same file?

I think there's still value to support go-to-definition in the same test file, which would require us either not excluding them, or implement something else for same-file indexing. I think we can go with former first until it's proven too costly?

Reply:
Indexing all test files does increase the time to index Core significantly. But also, it will be annoying once we add autocomplete since you almost never want results coming from tests unless you're inside that specific file.

I think we'll need to explore how to provide go to definition for things defined in the same file. But I wouldn't index the test files.

Reply:
At this time, it wouldn't impact Core since Sorbet can provide go to definition for any classes/modules even on `typed: false` files.

This would only impact Core once we add methods to the index. But when we get there, I agree that we need to handle same file definitions in a more specialized way.